src/neo/IO/Caching/HashSetCache.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Neo.IO.Caching
+{
+    public class HashSetCache<T> : IEnumerable<T> where T : IEquatable<T>
+    {
+        private readonly int hashSetCapacity;
+        private readonly List<HashSet<T>> sets = new List<HashSet<T>>();
+
+        public int Count
+        {
+            get
+            {
+                int count = 0;
+                foreach (var set in sets)
+                {
+                    count += set.Count;
+                }
+                return count;
+            }
+        }
+
+        public HashSetCache(int hashSetCapacity, int hashSetCount = 10)
+        {
+            if (hashSetCapacity <= 0) throw new ArgumentOutOfRangeException(nameof(hashSetCapacity));
+            if (hashSetCount <= 0 || hashSetCount > 20) throw new ArgumentOutOfRangeException($"{nameof(hashSetCount)} should between 1 and 20");
+
+            this.hashSetCapacity = hashSetCapacity;
+            for (int i = 0; i < hashSetCount; i++)
+            {
+                sets.Add(new HashSet<T>());
+            }
```

Comment:
`hashSetCapacity` and `hashSetCount` are confusing, as the class name is `HashSetCache`.  The `hashSetCapacity` is like the size of the `HashSetCache`.   Better to name them `bucketSize` and `bucketAmount` or other names.